File: skills/astra-toolkit/clients/java/examples/collections-find-many-sort-vector.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.cursor.CollectionFindCursor;
import com.datastax.astra.client.collections.commands.options.CollectionFindOptions;
import com.datastax.astra.client.collections.definition.documents.Document;
import com.datastax.astra.client.core.query.Sort;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient(System.getenv("ASTRA_DB_APPLICATION_TOKEN"))
            .getDatabase(System.getenv("ASTRA_DB_API_ENDPOINT"))
            .getCollection("**COLLECTION_NAME**");

    // Find documents
    CollectionFindOptions options =
        new CollectionFindOptions()
            .sort(Sort.vectorize("Text to vectorize"))
            .includeSortVector(true);
    CollectionFindCursor<Document, Document> cursor = collection.find(options);

    // Get the sort vector from the result
    System.out.println(cursor.getSortVector());
  }
}

File: skills/astra-toolkit/clients/typescript/examples/collections-find-many-sort-vector.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db(process.env.ASTRA_DB_API_ENDPOINT!, {
  token: process.env.ASTRA_DB_APPLICATION_TOKEN!,
});
const collection = database.collection("**COLLECTION_NAME**");

// Find documents
(async function () {
  const cursor = collection.find(
    {},
    {
      sort: { $vectorize: "Text to vectorize" },
      includeSortVector: true,
    },
  );

  // Get the sort vector from the result
  const vector = await cursor.getSortVector();
  console.log(vector);
})();
